pkg/platform/abstract/platform.go

```diff
@@ -1686,11 +1686,35 @@ func (ap *Platform) validateTriggers(functionConfig *functionconfig.Config) erro
 				}
 			}
 		}
+
+		if err := ap.validateBatchConfiguration(triggerInstance.Batch, functionConfig.Spec.Runtime, triggerInstance.Kind); err != nil {
+			return nuclio.WrapErrBadRequest(err)
+		}
 	}
 
 	return nil
 }
 
+func (ap *Platform) validateBatchConfiguration(batchConfiguration *functionconfig.BatchConfiguration, runtime, triggerKind string) error {
+	if batchConfiguration == nil {
+		return nil
+	}
+
+	if functionconfig.BatchModeEnabled(batchConfiguration) && (triggerKind != "http" || !strings.Contains(runtime, "python")) {
+		ap.Logger.WarnWith("Batching is not supported for given runtime and kind - batching configuration is ignored",
+			"runtime", runtime,
+			"triggerKind", triggerKind)
+	}
```

Comment:
I'd split it to make it clearer for the user:
```suggestion
	if functionconfig.BatchModeEnabled(batchConfiguration) {
	    if triggerKind != "http" {
	        ap.Logger.WarnWith("Batching is not supported for trigger kind - batching configuration is ignored",
			"triggerKind", triggerKind)
	    }
	    if !strings.Contains(runtime, "python")) {
	        ap.Logger.WarnWith("Batching is not supported for given runtime - batching configuration is ignored",
			"runtime", runtime)
	    }
	}
```

Reply:
Might be even better to specific functions `TriggerKindSupportsBatching` and `RuntimeSupportsBatching`, which we'll be able to extend

Reply:
@TomerShor good idea!